File: GamePlayBackend/src/main/kotlin/com/gameplay/gameplaybackend/models/GameRoomModel.kt
package com.gameplay.gameplaybackend.models

import com.fasterxml.jackson.annotation.JsonManagedReference
import jakarta.persistence.*

@Entity
@Table(name = "gameroom")
class GameRoomModel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id:Long = 0

    @Column( name = "id_category",nullable = false)
    var idCategory:Long = 0

    @Column(name = "id_game", nullable = false)
    var idGame:Long = 0

    @Column(name = "id_user_owner", nullable = false)
    var idUserOwner:Long = 0

    @Column(name = "date", nullable = false)
    var date:String = ""

    @Column(name = "description", nullable = false)
    var description:String = ""


    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_category", referencedColumnName = "id", insertable = false, updatable = false)
    val category:CategoryModel? = null

    @OneToMany(mappedBy = "gameRoom",fetch = FetchType.EAGER)
    var listPlayers: List<GameRoomPlayersModel> = mutableListOf()

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_user_owner", referencedColumnName = "id", insertable = false, updatable = false)
    var userOwner:UserModel? = null

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "id_game", referencedColumnName = "id", insertable = false, updatable = false)
    val game:GameModel? = null
}

File: gameplay_app/lib/src/features/home/domain/models/game_room_model.dart
import 'package:gameplay_app/src/features/home/domain/models/category_model.dart';
import 'package:gameplay_app/src/features/home/domain/models/game_model.dart';
import 'package:gameplay_app/src/features/home/domain/models/user_model.dart';

import 'game_room_players_model.dart';

class GameRoomModel {
  String id;
  String date;
  String description;
  String idGame;
  CategoryModel category;
  UserModel userOwner;
  GameModel game;
  List<GameRoomPlayersModel> listPlayers;

  GameRoomModel({
    required this.id,
    required this.date,
    required this.description,
    required this.idGame,
    required this.category,
    required this.userOwner,
    required this.game,
    required this.listPlayers,
  });

  factory GameRoomModel.fromMap(Map<String, dynamic> map) {
    return GameRoomModel(
      id: map['id'].toString(),
      date: map['date'],
      description: map['description'],
      idGame: map['idGame'].toString(),
      category: CategoryModel.fromMap(map['category']),
      userOwner: UserModel.fromMap(map['userOwner']),
      game: GameModel.fromMap(map['game']),
      listPlayers: List<GameRoomPlayersModel>.from(
          map['listPlayers']?.map((x) => GameRoomPlayersModel.fromMap(x))),
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'date': date,
      'description': description,
      'idGame': idGame,
      'category': category.toMap(),
      'userOwner': userOwner.toMap(),
      'game': game.toMap(),
      'listPlayers': listPlayers.map((x) => x.toMap()).toList(),
    };
  }
}
